fish-rust/src/wchar.rs

```diff
@@ -78,3 +42,9 @@ pub fn wchar_literal_byte(byte: u8) -> char {
     char::from_u32(u32::from(ENCODE_DIRECT_BASE) + u32::from(byte))
         .expect("private-use codepoint should be valid char")
 }
+
+pub fn wchar_decode_byte(c: char) -> u8 {
+    (u32::from(c) - u32::from(ENCODE_DIRECT_BASE))
+        .try_into()
+        .unwrap()
```

Comment:
Can we make this function return an `Option`? That way it can be called with any character, and the call site doesn't have to do checks manually (`if c >= ENCODE_DIRECT_BASE && c < ENCODE_DIRECT_END { wchar_decode_byte(c) }` becomes `if let Some(c) = wchar_decode_byte(c)`).